include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -8142,29 +8220,71 @@ FEFaceEvaluation<dim,
+        this->is_interior_face == false)
+      {
+        const auto cells    = this->get_cell_ids();
+        const auto face_nos = compute_face_no_data();
+
+        // TODO!
+        std::array<unsigned int, VectorizedArrayType::size()> face_orientations;
+        face_orientations.fill(0);
+
+        return internal::FEFaceEvaluationSelector<dim,
+                                                  fe_degree,
+                                                  n_q_points_1d,
+                                                  n_components,
+                                                  Number,
+                                                  VectorizedArrayType>::
+          gather_evaluate(input_vector.begin(),
+                          *this->data,
+                          *this->dof_info,
+                          this->begin_values(),
+                          this->begin_gradients(),
+                          this->scratch_data,
+                          evaluation_flag & EvaluationFlags::values,
+                          evaluation_flag & EvaluationFlags::gradients,
+                          this->active_fe_index,
+                          this->first_selected_component,
+                          cells,
+                          face_nos,
+                          this->subface_index,
+                          this->dof_access_index,
+                          face_orientations,
+                          this->mapping_data->descriptor[this->active_fe_index]
+                            .face_orientations);
+      }
+    else
+      {
+        return internal::FEFaceEvaluationSelector<dim,
+                                                  fe_degree,
+                                                  n_q_points_1d,
+                                                  n_components,
+                                                  Number,
+                                                  VectorizedArrayType>::
+          gather_evaluate(input_vector.begin(),
+                          *this->data,
+                          *this->dof_info,
+                          this->begin_values(),
+                          this->begin_gradients(),
+                          this->scratch_data,
+                          evaluation_flag & EvaluationFlags::values,
+                          evaluation_flag & EvaluationFlags::gradients,
+                          this->active_fe_index,
+                          this->first_selected_component,
+                          std::array<unsigned int, 1>{{this->cell}},
+                          std::array<unsigned int, 1>{{this->face_no}},
+                          this->subface_index,
+                          this->dof_access_index,
+                          std::array<unsigned int, 1>{{this->face_orientation}},
+                          this->mapping_data->descriptor[this->active_fe_index]
+                            .face_orientations);
+      }
+  };
+
+  if (!fu())
```

Comment:
This is where the implementations are called with `std::array`s of different lengths. 

@kronbichler This is what I was thinking of a couple of days ago. What do you think about it?